### PR TITLE
Add a pre-publish check to publish.

### PR DESCRIPTION
**Pull request by jeresig**
Also add a new `dry-run-publish` command so that we can test the publishing process.

Here is what the command is link and the kinds of warnings that you now get:

```
$ npm run dry-run-publish

> wonder-blocks@0.0.1 dry-run-publish /Users/jeresig/khan/wonder-blocks
> yarn run publish --no-npm --no-git

yarn run v1.7.0
$ node utils/pre-publish-check.js && git diff --stat --exit-code HEAD && yarn build:all && lerna publish --no-npm --no-git
@khanacademy/wonder-blocks-dropdown is private and won't be published to NPM.
@khanacademy/wonder-blocks-form is private and won't be published to NPM.
@khanacademy/wonder-blocks-grid is private and won't be published to NPM.
@khanacademy/wonder-blocks-toolbar is private and won't be published to NPM.
@khanacademy/wonder-blocks-tooltip is private and won't be published to NPM.
? There are some potential problems, do you wish to continue? (Y/n)
```

If you answer "no" then it errors out and stops the publish:

```
$ npm run dry-run-publish

> wonder-blocks@0.0.1 dry-run-publish /Users/jeresig/khan/wonder-blocks
> yarn run publish --no-npm --no-git

yarn run v1.7.0
$ node utils/pre-publish-check.js && git diff --stat --exit-code HEAD && yarn build:all && lerna publish --no-npm --no-git
@khanacademy/wonder-blocks-dropdown is private and won't be published to NPM.
@khanacademy/wonder-blocks-form is private and won't be published to NPM.
@khanacademy/wonder-blocks-grid is private and won't be published to NPM.
@khanacademy/wonder-blocks-toolbar is private and won't be published to NPM.
@khanacademy/wonder-blocks-tooltip is private and won't be published to NPM.
? There are some potential problems, do you wish to continue? No
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! wonder-blocks@0.0.1 dry-run-publish: `yarn run publish --no-npm --no-git`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the wonder-blocks@0.0.1 dry-run-publish script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jeresig/.npm/_logs/2018-08-06T17_43_39_855Z-debug.log
```

Additionally, it checks for the missing publishConfirm and errors out if it doesn't see it:

```
$ npm run dry-run-publish

> wonder-blocks@0.0.1 dry-run-publish /Users/jeresig/khan/wonder-blocks
> yarn run publish --no-npm --no-git

yarn run v1.7.0
$ node utils/pre-publish-check.js && git diff --stat --exit-code HEAD && yarn build:all && lerna publish --no-npm --no-git
ERROR: @khanacademy/wonder-blocks-dropdown is missing a "publishConfig": {"access": "public"} section.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! wonder-blocks@0.0.1 dry-run-publish: `yarn run publish --no-npm --no-git`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the wonder-blocks@0.0.1 dry-run-publish script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jeresig/.npm/_logs/2018-08-06T17_44_33_361Z-debug.log
```

We can add more checks later but these are a couple that've caused issues for me!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,8 +81,9 @@
   },
   "dependencies": {
     "aphrodite": "^1.2.5",
-    "prop-types": "^15.6.2",
+    "inquirer": "^6.0.0",
```
Comment (jeresig) [inline package.json]: This is the NPM module that is used by Lerna, I like it!

File: packages/wonder-blocks-dropdown/package.json [hunk 2]
```diff
@@ -10,6 +10,9 @@
   },
   "author": "",
   "license": "MIT",
+  "publishConfig": {
```
Comment (jeresig) [inline packages/wonder-blocks-dropdown/package.json]: We had two packages with missing publishConfigs! Whoops!

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit d9d3e51be224806fcdf5d6e001059de3aec3b12d

https://deploy-preview-264--wonder-blocks.netlify.com

File: utils/pre-publish-check.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                console.warn(
+                    `${name} is private and won't be published to NPM.`,
+                );
+                warnings = true;
+            }
+        }
+
+        if (warnings) {
+            inquirer
+                .prompt([
```
Comment (kevinbarabash) [inline utils/pre-publish-check.js]: Cool!

File: utils/pre-publish-check.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                // eslint-disable-next-line no-console
+                console.error(
+                    `ERROR: ${name} is missing a "publishConfig": {"access": "public"} section.`,
+                );
+                process.exit(1);
+            }
+
+            if (pkgJson.private) {
+                // eslint-disable-next-line no-console
+                console.warn(
```
Comment (kevinbarabash) [inline utils/pre-publish-check.js]: Yes!

File: utils/pre-publish-check.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            }
+        }
+
+        if (warnings) {
+            inquirer
+                .prompt([
+                    {
+                        type: "confirm",
+                        name: "skipWarnings",
+                        default: true,
```
Comment (kevinbarabash) [inline utils/pre-publish-check.js]: We should probably change this to `default: false`.

**Comment by jeresig** [comment]
@kevinbarabash Fixed!

**Pull request merged**